plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/Fill.java

```diff
@@ -148,4 +187,33 @@ public String getFillPattern() {
   public void setFillPattern(String value) {
     jsonObject.addProperty("fill-pattern", value);
   }
+
+  @Override
+  @Nullable
+  Geometry getOffsetGeometry(@NonNull Projection projection, @NonNull MoveDistancesObject moveDistancesObject,
+                             float touchAreaShiftX, float touchAreaShiftY) {
+    List<List<Point>> originalPoints = ((Polygon) getGeometry()).coordinates();
+    if (originalPoints != null) {
+      List<List<Point>> resultingPoints = new ArrayList<>(originalPoints.size());
+      for (List<Point> points : originalPoints) {
+        List<Point> innerPoints = new ArrayList<>();
+        for (Point jsonPoint : points) {
+          PointF pointF = projection.toScreenLocation(new LatLng(jsonPoint.latitude(), jsonPoint.longitude()));
+          pointF.x -= moveDistancesObject.getDistanceXSinceLast();
+          pointF.y -= moveDistancesObject.getDistanceYSinceLast();
+
+          LatLng latLng = projection.fromScreenLocation(pointF);
+          if (latLng.getLatitude() > MAX_MERCATOR_LATITUDE || latLng.getLatitude() < MIN_MERCATOR_LATITUDE) {
+            return null;
+          }
+          innerPoints.add(Point.fromLngLat(latLng.getLongitude(), latLng.getLatitude()));
+        }
+        resultingPoints.add(innerPoints);
+      }
+
+      return Polygon.fromLngLats(resultingPoints);
+    }
+
+    return null;
```

Comment:
do we want nullability for this? Returning an empty list would suffice and would make the method `@NonNull`

Reply:
In this case, we are returning `null` to indicate that the resulting geometry is invalid and we should not set it, leaving the old one being displayed